### PR TITLE
feat: add i18n for scaffolder

### PR DESCRIPTION
**Pull request by mario-mui**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mario-mui** [comment]
finish PR: https://github.com/backstage/backstage/pull/20577

**Comment by mario-mui** [comment]
ping @benjdlambert @freben

**Comment by freben** [comment]
Hi! First quick glance - I see a conflict has crept in. Also this should be exported on /alpha, right?

**Comment by mario-mui** [comment]
> Hi! First quick glance - I see a conflict has crept in. Also this should be exported on /alpha, right?

yes

File: plugins/scaffolder/api-report-alpha.md [hunk 1]
```diff
@@ -47,6 +48,137 @@ export type FormProps = Pick<
...
+    readonly 'ongoingTask.subtitle': 'Task {{taskId}}';
+    readonly 'ongoingTask.pageTitle.hasTemplateName': 'Run of {{templateName}}';
+    readonly 'ongoingTask.pageTitle.noTemplateName': 'Scaffolder Run';
+    readonly 'ongoingTask.cancelButtonTitle': 'Cancel';
+    readonly 'ongoingTask.startOverButtonTitle': 'Start Over';
+    readonly 'ongoingTask.hideLogsButtonTitle': 'Hide Logs';
+    readonly 'ongoingTask.showLogsButtonTitle': 'Show Logs';
+    readonly 'templateTypePicker.title': 'Categories';
+    readonly 'templateEditorPage.title': 'Template Editor';
+    readonly 'templateEditorPage.subTitle': 'Edit, preview, and try out templates and template forms';
```
Comment (benjdlambert) [inline plugins/scaffolder/api-report-alpha.md:138]:
```suggestion
    readonly 'templateEditorPage.subtitle': 'Edit, preview, and try out templates and template forms';
```

File: plugins/scaffolder/api-report-alpha.md [hunk 2]
```diff
@@ -47,6 +48,137 @@ export type FormProps = Pick<
...
+    readonly 'templateEditorPage.customFieldExplorer.fieldForm.applyButtonTitle': 'Apply';
+    readonly 'templateEditorPage.customFieldExplorer.preview.title': 'Example Template Spec';
+    readonly 'templateEditorPage.templateEditorBrowser.closeConfirmMessage': 'Are you sure? Unsaved changes will be lost';
+    readonly 'templateEditorPage.templateEditorBrowser.saveIconTooltip': 'Save all files';
+    readonly 'templateEditorPage.templateEditorBrowser.reloadIconTooltip': 'Reload directory';
+    readonly 'templateEditorPage.templateEditorBrowser.closeIconTooltip': 'Close directory';
+    readonly 'templateEditorPage.templateEditorIntro.title': 'Get started by choosing one of the options below';
+    readonly 'templateEditorPage.templateEditorIntro.loadLocal.title': 'Load Template Directory';
+    readonly 'templateEditorPage.templateEditorIntro.loadLocal.description': 'Load a local template directory, allowing you to both edit and try executing your own template.';
+    readonly 'templateEditorPage.templateEditorIntro.loadLocal.unSupportsLoadTooltip': 'Only supported in some Chromium-based browsers';
```
Comment (benjdlambert) [inline plugins/scaffolder/api-report-alpha.md:158]:
```suggestion
    readonly 'templateEditorPage.templateEditorIntro.loadLocal.unsupportedTooltip': 'Only supported in some Chromium-based browsers';
```

**Review by benjdlambert — COMMENTED** [review]
Think you'll need to run the API reports for catalog too, maybe you can do that in another PR instead and make those changes there?

Some small key name changes I thought of, but think that this is looking OK!

**Comment by mario-mui** [comment]
> Think you'll need to run the API reports for catalog too, maybe you can do that in another PR instead and make those changes there?

ok

File: plugins/scaffolder/src/translation.ts [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+      },
+      repoUrlPickerRepoName: {
+        repository: {
+          title: 'Repositories Available',
+          inputTitle: 'Repository',
+          description: 'The name of the repository',
+        },
+      },
+    },
+    listTaskPage: {
```
Comment (benjdlambert) [inline plugins/scaffolder/src/translation.ts:142]: I think that you can condense `repoUrlPickerHost` and `repoUrlPickerRepoName` into one `repoUrlPicker` section and have `host` and `repository` under that instead.

**Review by benjdlambert — COMMENTED** [review]
Just a small thing on the `RepoUrlPicker` section

**Review by benjdlambert — APPROVED** [review]
@mario-mui I went ahead and fixed this up and rebased it so I think we should be good to go! Thanks a lot for your hard work here as always! :pray:

**Pull request merged**